### ISSUE TITLE
Audio engine

### ISSUE THREAD

**Issue by cxong** [cxong/cdogs-sdl#568]
C-Dogs already uses some audio processing features like #507 , #438 , #369 but these are coded in a way that's hard to maintain. It would be nice to use an audio engine instead.

Try http://sol.gfxile.net/soloud/

**Comment by stavrossk** [comment]
@cxong Yes, that would be great. It would save you some time to focus on more important things for the game.
